File: play-2.4.x/K-Modules/kmodules/K-Base/app/K/Aop/annotations/FreqencyLimit.java
package K.Aop.annotations;

import K.Aop.actions.FreqencyLimitAction;
import play.mvc.With;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by kk on 14/11/11.
 */
@With(FreqencyLimitAction.class)
@Target({ElementType.TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
public @interface FreqencyLimit {
    public String[] IncludedParams() default {};

    public int TimeOutInSec() default 5;

    public String ErrMsg() default "调用频繁,超过限制,请稍后重试";
}

File: play-kotlin-2.5.x/PlayKotlin/kmodules/K-Base/app/K/Aop/annotations/FreqencyLimit.kt
package K.Aop.annotations

import K.Aop.actions.FreqencyLimitAction
import play.mvc.With
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * Created by kk on 14/11/11.
 */
@With(FreqencyLimitAction::class)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FILE, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.RUNTIME)
annotation class FreqencyLimit(val IncludedParams: Array<String> = arrayOf(), val TimeOutInSec: Int = 5, val ErrMsg:

String = "调用频繁,超过限制,请稍后重试")
